src/vm/jitinterface.cpp

```diff
@@ -9174,6 +9174,12 @@ CORINFO_CLASS_HANDLE CEEInfo::getDefaultEqualityComparerClassHelper(CORINFO_CLAS
             case ELEMENT_TYPE_U4:
             case ELEMENT_TYPE_I8:
             case ELEMENT_TYPE_U8:
+            case ELEMENT_TYPE_BOOLEAN:
+            case ELEMENT_TYPE_CHAR:
+            case ELEMENT_TYPE_R4:
+            case ELEMENT_TYPE_R8:
```

Comment:
Is this going to make Enum.Equals and EqualityComparer to have a different behavior for NaNs and other non-finite or denormalized numbers?

I think it would be ok to change the behavior of `Enum.Equals` with float/double underlying types to be in sync with float/double; but it should be done across the board.

Reply:
The changes related top EqualityComparer and Comparer for enums look pretty independent on the rest of the changes in this PR. I would help to submit them in a separate smaller PR.

Reply:
I've removed the comparer changes from this PR.